go-controller/pkg/libovsdbops/db_object_types.go

```diff
@@ -158,3 +164,13 @@ var ACLEgressFirewall = newObjectIDsType(acl, EgressFirewallOwnerType, []Externa
 	// The only additional id we need is the index of the EgressFirewall.Spec.Egress rule.
 	RuleIndex,
 })
+
+var VirtualMachineDHCPOptions = newObjectIDsType(dhcpOptions, VirtualMachineOwnerType, []ExternalIDKey{
+	// cidr
+	ObjectNameKey,
+	// We can have multiple VMs with same CIDR they  may have different
+	// hostname
+	VirtualMachineKey,
+	// Also differente namespace so we have on DHCPOption per LSP
+	NamespaceKey,
```

Comment:
the VirtualMachineDHCPOptions key should be a tuple of cidr+namespace+vmName, the last question is if we can use a new Key like `CidrKey` instead of `ObjectNameKey`

Reply:
@npinaeva I discussed with @qinqon two other alternatives to what is being done here:

- Set `ObjectNameKey` to `<namespace name>/<vm-name> and have an additional `CIDRKey` set to `<cidr>`
- Not use `ObjectNameKey` altogether and have `CIDRKey` set to `<cidr>`

My argumentation here is that if the type is `VirtualMachineOwnerType`, the `ObjectNameKey`, if any, should relate and refer to and ID of the VM and VM name might be an ID that is easier to relate that the CIDR.

I also thought that `ObjectNameKey` was mandatory but maybe not?

But this all might be subjective. So wanted a second opinion.

Reply:
so `ObjectNameKey` is supposed to be a part of the secondary index, that is shared across the whole nbdb table (but we don't have secondary indexing implemented yet)
It is usually useful to put information that helps you to clean up everything related to a single object to this field, e.g. I see in this code delete is happening from `kubevirt/dhcp:DeleteDHCPOptions`. Looks like it deletes all dhcp options for the same vmNamespace+vmName (and we know all created cidrs should be deleted)
If that is correct, @jcaamano first option makes sense, and then you can have a predicate-based cleanup without the need to extract all pod ips. If you use `ObjectNameKey`, then when we implement secondary indexes, this predicate-search will become more efficient automatically :)